pkg/cluster/deploybaseresources.go

```diff
@@ -444,6 +454,60 @@ func (m *manager) setMasterSubnetPolicies(ctx context.Context) error {
 	return err
 }
 
+func (m *manager) generateFederatedIdentityCredentials(ctx context.Context) error {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil
+	}
+
+	if m.doc.OpenShiftCluster.Properties.ClusterProfile.OIDCIssuer == nil {
+		return fmt.Errorf("OIDCIssuer is nil for the cluster with ID: %s", m.doc.OpenShiftCluster.ID)
+	}
+
+	Issuer := to.StringPtr((string)(*m.doc.OpenShiftCluster.Properties.ClusterProfile.OIDCIssuer))
+
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		platformWIRole, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
```

Comment:
There can be some managed identities in  platformIdentities list(`m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities`) that won't be present in `m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()`. 

The `platformWIRolesByRoleName` should be used to confirm what identities are need to be federated, thus here whatever `exists` only those should be federated.

Reply:
so, if I understand correctly, we don't throw an error instead `if !exists { continue }` ?

Reply:
updated